File: Android/AddControl-src/AddControl.java
package com.magicsoftware.magicdev;

import com.magicsoftware.core.CoreApplication;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.content.Context;

public class AddControl {
	// get a value of a magic group control and add a child control to it
	public static void add(final String ctrlname, final int generation) {
		CoreApplication.getInstance().currentActivity.runOnUiThread(new Runnable() {
			@Override
			public void run() {
				// find the magic control
				View myCustomView = CoreApplication.getInstance().getControlByName(ctrlname,generation);
				if (myCustomView != null) {
					//defines a custom control
					TextView myTextView = new TextView(CoreApplication.getInstance().getApplicationContext());
					myTextView.setText("Hello World");
					//adds the custom controls to the magic group control (panel)
					((ViewGroup) myCustomView).addView(myTextView, 0, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
				}
			}
		});
	}
}

File: iOS/AddControl-MagicApp/AddControl.m
//
//  AddControl.m
//  MagicApp
//

#import "AddControl.h"
#import "Magicxpa.h"

@interface AddControl ()

@end

@implementation AddControl

+(void)add:(NSArray *)params
{
	// get a value of a magic group control and add a child control to it
	NSString *ctrlname = [params objectAtIndex: 0];
	long generation = [[params objectAtIndex:1] longValue];
	
	// find the magic control
	UIView *myCustomView = [Magicxpa getControlByName:ctrlname TaskGeneration:generation];
	// calls a native code to create a new native control and add it to the magic control
    [self performSelectorOnMainThread:@selector(addControl:) withObject:myCustomView waitUntilDone:YES];
}

+(void)addControl:(UIView *)control
{
	// defines a custom control
    UILabel *label = [[UILabel alloc] initWithFrame:control.bounds];
    [label setText:@"Hello World"];
    [label sizeToFit];
	// adds the custom controls to the magic group control
	[control addSubview:label];
}

@end
